configuration-rendering-controller/README.md

```diff
@@ -0,0 +1,481 @@
+generated CMs and Secrets in any namespace.
+
+For workload cluster `app-operators` that might need access to configurations in the future, we could either give
+access to the namespace for CMs and Secrets that contains all rendered configs - if they would use the same -
+or render the workload cluster specific level to the given namespace. For one, the CR should be able to support that
+scope change. For another we do not have that implemented in the current system. The main issue here however is not
+a problem with this system, but with what we can share freely with customer workloads. For example, `shared-configs`
+is still kept private, because it contains - even if each encrypted with MC unique keys - minor Giant Swarm sensitive
+information. They are required by vintage to my knowledge, but they are there nevertheless and a security audit
+should be carried before we make that repository public.
+
+### Extending configuration management in the future
+
+Currently, we mostly utilise Catalog configuration for cluster apps. I think the reason for it is that it is not
+possible at the moment to utilise the configuration system when we deploy such workloads, so the configuration
+drifted into the catalogs, as the `cluster` catalog has well-defined scope for such apps.
+
+With easy access to configuration management, these catalog configs could simply drift into the config structure
+for the apps that need them on the MC specific level, and we could then simply get rid of them.
+
+## Implementation approaches
+
+### Framework
+
+The new controller should be written using [kubebuilder](https://book.kubebuilder.io/). It is on our roadmap to
+[Remove `operatorkit` as a dependency](https://github.com/giantswarm/roadmap/issues/3722) and this would be a great
+opportunity to get some hands-on experience with `kubebuilder`, as most of those operators that need and upgrade
+is owned by Team HoneyBadger.
+
+### Possibility to utilize Flux source-controller
+
+It is already implemented in `konfigure` to fetch the sources from Flux `source-controller` with `include`
+feature support [here](https://github.com/giantswarm/konfigure/pull/278).
+
+It would potentially be best to not depend on Flux either, but since we already have it implemented, and we would
+need to do something very similar to fetch and assemble the sources, we could cut this corner now and keep
+using Flux.
+
+There is potential to later implement custom DSL to describe how to merge configurations, which could make sense
+if we want to introduce many different layers in the future (catalogs, workload-cluster, app instance, etc. specific
+configs). For now, it makes sense to keep our current logic as is to reduce time to implement and friction.
+
+### Defining scope of configs to render
+
+```yaml
+apiVersion: configuration.giantswarm.io/v1alpha1
+kind: ManagementClusterConfiguration
```

Comment:
Why do we need a ManagementClusterConfiguration and not a simple Configuration CRD?

Reply:
We potentially want to extend the layers the configuration system can render that are currently not possible but there is a need for it, e.g. catalog configs, workload cluster configs, app instance configs (same app deployed multiple times and you want different configurations for them).

A single CRD may not be suitable to cram all that into it. We aim to cover what we currently do with the config system and that is technically configuration for MC apps. Therefor a name that represents that feels more future proof.

Reply:
I would have thought this could be done using a hierarchical system of configs as well.

You could have:
- a main config
- a mc specific config
- an app instance config

And the apps would point to a base config so this would be as flexible but less specific in semantics?